kasa/smart/modules/temperaturecontrol.py

```diff
@@ -48,6 +88,45 @@ def query(self) -> dict:
         # Target temperature is contained in the main device info response.
         return {}
 
+    @property
+    def state(self) -> bool:
+        """Return thermostat state."""
+        return self._device.sys_info["frost_protection_on"] is False
+
+    async def set_state(self, enabled: bool):
+        """Set thermostat state."""
+        return await self.call("set_device_info", {"frost_protection_on": not enabled})
+
+    @property
+    def mode(self) -> ThermostatState:
+        """Return thermostat state."""
+        # If frost protection is enabled, the thermostat is off.
```

Comment:
Is this definitely the case? Can't both be on even though the frost alarm shouldn't trigger

Reply:
So I think it just sets the minimum allowed temperature and turns heating on if that's reached. I don't think there's a separate way to turn it off, but we can improve on that later.